src/sage/combinat/designs/covering_array.py

```diff
@@ -49,3 +54,216 @@
+    If so return the CA, if not return False.
+
+    INPUT:
+
+    - ``strength`` (integer) -- the parameter `t` of the covering array,
+      such that in any selection of `t` columns of the array, every
+      `t`-tuple appears at least once.
+
+    - ``levels`` (integer) -- the parameter `v` which is the number of
+      unique symbols that appear in the covering array.
+
+    - ``number_columns`` (integer) -- the number of columns desired for
+      the covering array.
+
+    EXAMPLES::
+
+        sage: from sage.combinat.designs.designs_pyx import is_covering_array
+        sage: from sage.combinat.designs.covering_array import database_check
+        sage: C = database_check(6, 2, 3)
+        sage: is_covering_array(C, parameters=True)
+        (True, (12, 2, 6, 3))
+        sage: database_check(6, 3, 3)
+        False
+
+    """
+    import sage.combinat.designs.database as DB
+
+    if (strength, levels) in DB.CA_constructions:
+        for i in DB.CA_constructions[(strength, levels)]:
+            if number_columns <= i[1]:
+                CA = "ca_{}_{}_{}_{}".format(i[0], strength, i[1], levels)
+                f = getattr(DB, CA)
+                return truncate_columns(f(), number_columns)
+        return False
+    else:
+        return False
+
+
+def covering_array(strength, number_columns, levels):
+    r"""
+    Build a `CA(N; t, k, v)` using direct constructions, where `N` is the
+    smallest size known.
+
+    INPUT:
+
+    - ``strength`` (integer) -- the parameter `t` of the covering array,
+      such that in any selection of `t` columns of the array, every
+      `t`-tuple appears at least once.
+
+    - ``levels`` (integer) -- the parameter `v` which is the number of
+      unique symbols that appear in the covering array.
+
+    - ``number_columns`` (integer) -- the number of columns desired for
+      the covering array.
+
+    EXAMPLES::
+
+        sage: from sage.combinat.designs.designs_pyx import is_covering_array
+        sage: from sage.combinat.designs.covering_array import covering_array
+        sage: C1 = covering_array(2, 7, 3)
+        sage: is_covering_array(C1,parameters=True)
+        (True, (12, 2, 7, 3))
+        sage: C2 = covering_array(2, 11, 2)
+        sage: is_covering_array(C2,parameters=True)
+        (True, (7, 2, 11, 2))
+        sage: C3 = covering_array(2, 8, 7)
+        sage: is_covering_array(C3,parameters=True)
+        (True, (49, 2, 8, 7))
+        sage: C4 = covering_array(2, 50, 7)
+        No direct construction known and/or implemented for a CA(N; 2, 50, 7)
+
+    """
+    from sage.combinat.designs.orthogonal_arrays import orthogonal_array
+
+    if levels == 2 and strength == 2:
+        return column_Kleitman_Spencer_Katona(number_columns)
+
+    in_database = database_check(number_columns, strength, levels)
+    if in_database:
+        return in_database
+
+    if orthogonal_array(number_columns, levels, strength, existence=True) == True:
```

Comment:
see message from Ruff

Reply:
https://github.com/sagemath/sage/pull/37372#issuecomment-2174348903

I noticed this but I am not sure how to proceed to make it happy while also keeping the functionality we need.

Reply:
```suggestion
    if orthogonal_array(number_columns, levels, strength, existence=True) is True:
```